lib/admin-ui/src/utils/time.js

```diff
@@ -0,0 +1,79 @@
+// From:
+// https://muffinman.io/blog/javascript-time-ago-function/
+
+const MONTH_NAMES = [
+  "January",
+  "February",
+  "March",
+  "April",
+  "May",
+  "June",
+  "July",
+  "August",
+  "September",
+  "October",
+  "November",
+  "December",
+];
+
+// --- Main function
+export function timeAgo(dateParam) {
+  if (!dateParam) {
+    return null;
+  }
+
+  const date = typeof dateParam === "object" ? dateParam : new Date(dateParam);
+  const DAY_IN_MS = 86400000; // 24 * 60 * 60 * 1000
+  const today = new Date();
+  const yesterday = new Date(today - DAY_IN_MS);
+  const seconds = Math.round((today - date) / 1000);
+  const minutes = Math.round(seconds / 60);
+  const isToday = today.toDateString() === date.toDateString();
+  const isYesterday = yesterday.toDateString() === date.toDateString();
+  const isThisYear = today.getFullYear() === date.getFullYear();
+
+  if (seconds < 5) {
+    return "now";
+  } else if (seconds < 60) {
+    return `${seconds} seconds ago`;
+  } else if (seconds < 90) {
+    return "about a minute ago";
+  } else if (minutes < 60) {
+    return `${minutes} minutes ago`;
+  } else if (isToday) {
+    return getFormattedDate(date, "Today"); // Today at 10:20
+  } else if (isYesterday) {
+    return getFormattedDate(date, "Yesterday"); // Yesterday at 10:20
+  } else if (isThisYear) {
+    return getFormattedDate(date, false, true); // 10. January at 10:20
+  }
+
+  return getFormattedDate(date); // 10. January 2017. at 10:20
+}
+
+function getFormattedDate(date, prefomattedDate = false, hideYear = false) {
```

Comment:
Do you prefer using utils like `timeAgo` instead of using e.g. `formatDistanceToNow` from `date-fns`?

Reply:
Yes, in this case, I prefer adding a few lines of code that I can quickly scan and understand, over importing a large library. I know we will rewrite the admin sooner rather than later, so a quick import of just the function I need is less overhead to me.

While I generally prefer a short snippet over importing a package (that might import another 50 sub-packages itself), I must admit, date-fns seems well designed, tree-shakable, without dependencies, and it will definitely be something I will use next time around. 🙌